connaisseur/image.py

```diff
@@ -27,61 +27,63 @@ class Image:
     tag: Optional[str]
     digest: Optional[str]
 
-    def __init__(self, image: str):
-        separator = r"[-._:@+]|--"
-        alphanum = r"[A-Za-z0-9]+"
-        component = f"{alphanum}(?:(?:{separator}){alphanum})*"
-        ref = f"^{component}(?:/{component})*$"
-
-        # e.g. :v1, :3.7-alpine, @sha256:3e7a89...
-        tag_re = r"(?:(?:@sha256:([a-f0-9]{64}))|(?:\:([\w.-]+)))"
-
-        match = re.search(ref, image)
-        if not match:
-            msg = "{image} is not a valid image reference."
-            raise InvalidImageFormatError(message=msg, image=image)
-
-        name_tag = image.split("/")[-1]
-        search = re.search(tag_re, name_tag)
-        self.digest, self.tag = search.groups() if search else (None, "latest")
-        self.name = name_tag.removesuffix(":" + str(self.tag)).removesuffix(
-            "@sha256:" + str(self.digest)
+    def __init__(self, image: str):  # pylint: disable=too-many-locals
```

Comment:
This seems like a larger rework. Would add tests for the changed/expected behaviour. E.g. afai can see, you added the length check but never tested it

Reply:
i added tests for the to long name and non supported digest algo.